File: Other/TimeConversion.py
from nose.tools import assert_equal

def time_Con(time):
	if time[8:] == "PM":
		if time[:2] == "12":
			return time[:8]
		else:
			hour = int(time[:2]) + 12
			return str(hour) + time[2:8]
	else:
		if time[:2] == "12":
			return "00" + time[2:8]
		else:
			return time[:8]

class Time_Con(object):
	def test(self, func):
		assert_equal(func("12:00:00AM"), "00:00:00")
		assert_equal(func("01:50:00AM"), "01:50:00")
		assert_equal(func("06:25:10AM"), "06:25:10")
		assert_equal(func("12:00:00PM"), "12:00:00")
		assert_equal(func("02:00:00PM"), "14:00:00")
		assert_equal(func("07:59:00PM"), "19:59:00")
		assert_equal(func("11:33:59PM"), "23:33:59")
		print("TESTS PASSED")

t = Time_Con()
t.test(time_Con)

File: Other/TimeConversion.js
function timeCon(time){
	if(time.substring(8, 10) === "PM"){
		if(time.substring(0, 2) === "12"){
			console.log(time.substring(0, 8));
		} else{
			var hour = parseInt(time.substring(0, 2)) + 12;
			console.log(hour.toString() + time.substring(2, 8));
		}
	} else{
		if(time.substring(0, 2) === "12"){
			console.log("00" + time.substring(2, 8));
		} else{
			console.log(time.substring(0, 8));
		}
	}
}

timeCon("12:10:50AM");
timeCon("01:02:49AM");
timeCon("11:00:10AM");
timeCon("12:00:00PM");
timeCon("06:38:19PM");